### PR TITLE
expression: implement vectorized evaluation for 'builtinSubDurationAndStringSig' 

### PR DESCRIPTION
**Pull request by SilvaXiang**
### What problem does this PR solve? 
implement vectorized evaluation for 'builtinSubDurationAndStringSig' from #12176 

### What is changed and how it works?

7.7% faster than before:

    BenchmarkVectorizedBuiltinTimeFunc/builtinSubDurationAndStringSig-VecBuiltinFunc-4                   800           1491651 ns/op           94184 B/op       5814 allocs/op
    BenchmarkVectorizedBuiltinTimeFunc/builtinSubDurationAndStringSig-NonVecBuiltinFunc-4                754           1617443 ns/op           94250 B/op       5814 allocs/op

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -1203,11 +1203,53 @@ func (b *builtinToSecondsSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+	result.MergeNulls(buf1)
+	arg0s := buf0.GoDurations()
+	resultSlice := result.GoDurations()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		// calculate
```
Comment (b41sh) [inline expression/builtin_time_vec.go]: These comments are meaningless, we can remove it.

Comment (SilvaXiang) [reply expression/builtin_time_vec.go]: OK，I have  changed it .

**Comment by b41sh** [comment]
LGTM

**Pull request closed without merging**